homeassistant/components/sms/notify.py

```diff
@@ -8,30 +8,36 @@
 from homeassistant.const import CONF_NAME, CONF_RECIPIENT, CONF_TARGET
 import homeassistant.helpers.config_validation as cv
 
-from .const import DOMAIN, GATEWAY, SMS_GATEWAY
+from .const import CONF_UNICODE, DOMAIN, GATEWAY, SMS_GATEWAY
 
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_RECIPIENT): cv.string, vol.Optional(CONF_NAME): cv.string}
+    {vol.Optional(CONF_RECIPIENT): cv.string, vol.Optional(CONF_NAME): cv.string}
 )
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the SMS notification service."""
 
+    return SMSNotificationService(hass)
+
+
+async def async_get_service(hass, config, discovery_info=None):
+    """Get the SMS notification service."""
+
     if discovery_info is None:
-        number = config[CONF_RECIPIENT]
+        number = config.get(CONF_RECIPIENT, None)
     else:
-        number = discovery_info[CONF_RECIPIENT]
+        number = discovery_info.get(CONF_RECIPIENT, None)
```

Comment:
We can outdent here after removing the else block.

Reply:
We don't seem to pass the number in the discovery info.

Reply:
> We don't seem to pass the number in the discovery info.

Can you elborate? Are you saying that there is a bug here?

Reply:
Yes, we do not set `CONF_RECIPIENT` in the discovery info when loading the notify platform. The discovery info is the fourth parameter to `discovery.async_load_platform`.

Reply:
But, if I do this change, then the config from yaml does not work anymore. E.g.:
```
notify:
  - name: sms_user1
    platform: sms
    recipient: "+1XXXXXX"
  - name: sms_user2
    platform: sms
    recipient: "+1YYYYYYY"
```

Reply:
Yes, we shouldn't support config yaml anymore. That's legacy.